tests/cross/src/main/scala/tests/pc/BaseTypeDefinitionSuite.scala

```diff
@@ -0,0 +1,73 @@
+package tests.pc
+
+import tests.BasePCSuite
+import scala.concurrent.duration.Duration
+import scala.meta.internal.metals.CompilerOffsetParams
+import collection.JavaConverters._
+import org.eclipse.{lsp4j => l}
+
+class BaseTypeDefinitionSuite extends BasePCSuite {
+
+  override def beforeAll(): Unit = {
+    indexJDK()
+    indexScalaLibrary()
+  }
+
+  def locationFromString(str: String): l.Location = {
```

Comment:
Let's try to use the same testing infrastructure as "textDocument/definition" instead of coming up with a new expect syntax.

Reply:
In general, I have found it helpful to test "goto definition" by running it on all possible tokens instead of writing individual test cases for custom positions.

Reply:
Could you give me example of suite you have in mind?

Reply:
I was referring to something like `DefinitionLspSuite`

Reply:
https://github.com/scalameta/metals/blob/cf7b95265a6019a99b0dea323caef47096dd8a58/tests/unit/src/test/scala/tests/DefinitionLspSuite.scala#L59-L78

but it seems to only be available in lsp suites

Reply:
Instead of using that structure, we can reuse `PcDefinitionSuite`, see https://github.com/scalameta/metals/blob/cf7b95265a6019a99b0dea323caef47096dd8a58/tests/cross/src/test/scala/tests/pc/PcDefinitionSuite.scala#L52

Reply:
As a general rule of thumb, avoid hardcoding line:column numbers in expect tests. Numbers are harder review than something like "Int.scala"

Reply:
Ok, I see what you have in mind.

Reply:
I've changed cross tests for goto-type-definition, now they share code with cross tests for goto-definition.